### PR TITLE
Fix the peer reaper.

### PR DESCRIPTION
**Pull request by Raynos**
We had a bug in interval_scan; I added a test that the
peer reaper actually runs over real time.

r: @jcorbin @kriskowal @rf

### CODE REVIEW

File: lib/interval-scan.js [hunk 1]
```diff
@@ -68,6 +68,9 @@ function setInterval(interval) {
     var self = this;
 
     if (self.interval === interval) {
+        if (!self.timer) {
```
Comment (kriskowal) [inline lib/interval-scan.js]: You can rely on setTimer to early return if !self.timer.

Comment (jcorbin) [inline lib/interval-scan.js]: :+1: , just call self.setTImer()

File: test/register/peer-reaper-runs.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var setTimeout = require('timers').setTimeout;
+
+var allocCluster = require('../lib/test-cluster.js');
+
+allocCluster.test('can register', {
```
Comment (jcorbin) [inline test/register/peer-reaper-runs.js]: should change this to be more descriptive

**Comment by jcorbin** [comment]
lgtm now

**Comment by kriskowal** [comment]
:ok:

**Comment by jcorbin** [comment]
The regression was caused by #96 which generalized the pattern used by the peer reaper, which we then went on to use in #100, #101, and the upcoming #116.

**Pull request merged**